tests/phpunit/integration/Modules/AdSense/SettingsTest.php

```diff
@@ -59,6 +59,7 @@ public function test_get_default() {
 				'accountSetupComplete' => false,
 				'siteSetupComplete'    => false,
 				'useSnippet'           => true,
+				'ownerID'              => 0,
```

Comment:
We should also test that the `ownerID` is set and updated accordingly when the owned keys are added/changed.

Rather than duplicate these for all module settings tests that have owned keys, we should add a contract test  trait that we can `use` in the respective `ModuleSettings` sub-class test cases. See https://github.com/google/site-kit-wp/blob/develop/tests/phpunit/includes/Core/Modules/Module_With_Settings_ContractTests.php for an example of one of these traits.

Reply:
Updated, but decided not to use trait since tests for different modules rely on different keys.

Reply:
After looking over all the added tests here, I still think it makes sense to use an abstraction for these tests since the tests are essentially all the same with different keys. Using an abstraction will allow us to provide more robust coverage (i.e. test every key) with less code, and a single place to manage it.

The test trait can have an abstract `get_owned_keys` method which returns the list of keys which are owned. In order to use the test trait, those keys need to be provided. The non-owned keys would be an array diff of this array with the keys from the setting's default value (whatever comes back from `get()`, the keys will always be the same). One alternative to having this method on the test trait would be to add the `get_owned_keys` method to the `Module_With_Owned_Keys` interface and that would make them easily available in the test as well, although I'm not sure if @felixarntz would approve of this change.

Let's implement the `get_owned_keys` on the test trait first, and if Felix is okay with moving it to the interface, then that is a very simple refactor to make later.

Reply:
I think we can expose these keys via the `Setting_With_Owned_Keys_Trait`, and let's introduce a `Setting_With_Owned_Keys_Interface` for that purpose, which exposes the public `get_owned_keys` method.

Reply:
Thanks for your input @felixarntz – I mixed up the interface name above; +1 on adding a `Setting_With_Owned_Keys_Interface`

There is the `Module_With_Owner` interface introduced in this PR along with the `Module_With_Owner_Trait` but neither are actually used yet.

> Regarding the trait, we should probably also make the name of the `ownerID` key a parameter so that it can be controlled by the class using the trait.

Sounds good. How about we make it an optional parameter to `Setting_With_Owned_Keys_Trait::register_owned_keys` with a default value of `ownerID` since we don't have any cases where it needs to be different?

@eugene-manuilov to summarize:
* The relevant module sub-classes need to implement the `Module_With_Owner` interface and `use` the `Module_With_Owner_Trait`
    - A `\Google\Site_Kit\Tests\Core\Modules\Module_With_Owner_ContractTests` trait should be added for adding test coverage to a Module sub-class' test case, and `use`d by the respective `Module` sub-class' test cases
* A `Setting_With_Owned_Keys_Interface` should be added with `get_owned_keys() : string[]` and the same module settings classes that use the trait should implement this interface as well
    - The contract tests trait for `Setting_With_Owned_Keys_ContractTests` will now have access to the list of owned keys directly from the class under test via the `get_owned_keys` method (rather than putting this on the test trait itself as mentioned previously)
* `Setting_With_Owned_Keys_Trait::register_owned_keys` should accept an optional `$ownerID_key = 'ownerID'` as the second parameter.

Reply:
Added.